expr/func_aspercent.go

```diff
@@ -131,8 +130,12 @@ func (s *FuncAsPercent) Exec(cache map[Req][]models.Series) ([]models.Series, er
 		if len(totals) == 1 {
 			totalsSerie = totals[0]
 		} else if len(totals) == len(series) {
-			sort.Sort(sortByTarget(series))
-			sort.Sort(sortByTarget(totals))
+			sort.Slice(series, func(i, j int) bool {
+				return series[i].Target < series[j].Target
+			})
```

Comment:
You could make a local func here like:
```
targetLess := func(i, j int) bool {
    return series[i].Target < series[j].Target
}
```

and reuse it for both sort calls.

Reply:
I can't because one is using series slice and the other is using totals

Reply:
Ah you are correct. I recommend leaving it as is, but you *could* do something cool with closures like:

```
targetLess := func(s []model.Series) func(i, j int) bool {
    return func(i, j int) bool {
        return s[i].Target < s[j].Target
    }
}

sort.Slice(series, targetLess(series))
sort.Slice(totals, targetLess(totals))
```

But I think this is harder to read. If you had to do it 5+ times or something this would make sense.

Reply:
Yeah that'd be nice if i had a lot of them. I'll keep it as is for this one